pkg/kedge/http/lbtransport/transport.go

```diff
@@ -111,15 +113,25 @@ func (s *tripper) RoundTrip(r *http.Request) (*http.Response, error) {
 	if irrecoverableErr != nil {
 		err := errors.Wrapf(irrecoverableErr, "lb: critical naming.Watcher error for target %s. Tripper is closed.", s.targetName)
 		reporter.Extract(r).ReportError(errtypes.IrrecoverableWatcherError, err)
+		_ = tripperware.Close(r.Body)
 		return nil, err
 	}
 
 	if len(targetsRef) == 0 {
 		err := errors.Errorf("lb: no backend is available for %s. 0 resolved addresses.", s.targetName)
 		reporter.Extract(r).ReportError(errtypes.NoResolutionAvailable, err)
+		_ = tripperware.Close(r.Body)
 		return nil, err
 	}
 
+	if r.Body != nil {
+		// We have to own true body for the request because we cannot reuse same reader closer
+		// in multiple calls to http.Transport.
```

Comment:
Isn't this what `http.Request`'s `GetBody` is for? i.e. use `GetBody` if available, replace with lazy buffer thingy if GetBody unavailable? might be too complex for now.

Reply:
This makes sense I think

Reply:
But we still have to have reference to `*lazyBufferedReader` to rewind so let's maybe stick to this now.